# HG changeset patch
# User Rik <rik@octave.org>
# Date 1430494192 25200
#      Fri May 01 08:29:52 2015 -0700
# Branch stable
# Node ID 4a41b7d426b604ca71df2d945168b502aae68b35
# Parent  f0a8c1bc3b6083e58b881d92699b91b1efe771ac
Fix loading of ASCII files with NA values in final column (bug #44967).

* lo-utils.cc (read_inf_nan_na): Check that last character read was not an EOF
indicator (-1) before attempting to put it back on the stream.

diff --git a/liboctave/util/lo-utils.cc b/liboctave/util/lo-utils.cc
--- a/liboctave/util/lo-utils.cc
+++ b/liboctave/util/lo-utils.cc
@@ -234,17 +234,18 @@ read_inf_nan_na (std::istream& is, char 
         if (c1 == 'a' || c1 == 'A')
           {
             char c2 = is.get ();
             if (c2 == 'n' || c2 == 'N')
               val = std::numeric_limits<T>::quiet_NaN ();
             else
               {
                 val = octave_numeric_limits<T>::NA ();
-                is.putback (c2);
+                if (c2 != EOF)
+                  is.putback (c2);
               }
           }
         else
           is.setstate (std::ios::failbit);
       }
       break;
 
     default:
